sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c

```diff
@@ -33,12 +85,9 @@ int main()
       az_span_from_str(getenv(CLIENT_ID_ENV)),
       az_span_from_str(getenv(CLIENT_SECRET_ENV)));
```

Comment:
This may destroy the span on line 85. A second `getenv` is not guaranteed to preserve the previously returned buffer.

Reply:
Similar to other iot samples, we need to copy the result of `getenv` into a buffer before passing it to key vault, to avoid the issue that @CIPop brought up. Let's fix this separately from this PR as it is an unrelated change.